chatmodels/openai/internal/openaiclient/chat.go

```diff
@@ -0,0 +1,79 @@
+package openaiclient
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"net/http"
+)
+
+// ChatPayloadMessage is a message in the format expected by the OpenAI API.
+type ChatPayloadMessage struct {
```

Comment:
as already said in the comments in https://github.com/tmc/langchaingo/issues/43 this format is currently not compatible with the LLM interface but is required by the openAI API.